drivers/storage/mountmgr/mountmgr.c

```diff
@@ -355,44 +348,39 @@ QueryDeviceInformation(IN PUNICODE_STRING SymbolicName,
         }
     }
 
-    /* If caller wants to know if there's valid contents */
-    if (Valid)
+    /* If caller wants to know if this is a FT volume */
+    if (IsFT)
     {
-        /* Suppose it's not OK */
-        *Valid = FALSE;
+        /* Suppose it's not */
+        *IsFT = FALSE;
 
+        /* FT volume can't be removable */
         if (!IsRemovable)
         {
-            /* Query partitions information */
+            /* Query partition information */
             Status = MountMgrSendSyncDeviceIoCtl(IOCTL_DISK_GET_PARTITION_INFO_EX,
                                                  DeviceObject,
                                                  NULL,
                                                  0,
                                                  &PartitionInfo,
                                                  sizeof(PartitionInfo),
                                                  NULL);
-#if 0
-            if (Status == STATUS_INSUFFICIENT_RESOURCES)
-            {
-                ObDereferenceObject(DeviceObject);
-                ObDereferenceObject(FileObject);
-                return Status;
-            }
-#endif
-            /* Once again here, failure isn't major */
+            /* Failure isn't major */
             if (!NT_SUCCESS(Status))
             {
                 Status = STATUS_SUCCESS;
             }
-            /* Verify we know something in */
-            else if (PartitionInfo.PartitionStyle == PARTITION_STYLE_MBR &&
-                     IsRecognizedPartition(PartitionInfo.Mbr.PartitionType))
+            /* Check if this is a FT volume */
+            else if ((PartitionInfo.PartitionStyle == PARTITION_STYLE_MBR) &&
```

Comment:
GCC moment: `error: suggest parentheses around '&&' within '||' [-Werror=parentheses]`

Reply:
Needs global parentheses around IsFTPartition definition in sdk/include/psdk/ntdddisk.h
```
#define IsFTPartition( PartitionType ) \
  (((PartitionType) & PARTITION_NTFT) && (((PartitionType) & ~0xC0) == PARTITION_FAT_12)) || \
  (((PartitionType) & PARTITION_NTFT) && (((PartitionType) & ~0xC0) == PARTITION_HUGE)) || \
  (((PartitionType) & PARTITION_NTFT) && (((PartitionType) & ~0xC0) == PARTITION_IFS)) || \
  (((PartitionType) & PARTITION_NTFT) && (((PartitionType) & ~0xC0) == PARTITION_FAT32)) || \
  (((PartitionType) & PARTITION_NTFT) && (((PartitionType) & ~0xC0) == PARTITION_FAT32_XINT13)) || \
  (((PartitionType) & PARTITION_NTFT) && (((PartitionType) & ~0xC0) == PARTITION_XINT13))
```

Reply:
Thanks, yes this is what was missing :+1:

Reply:
Fixed in master, and PR rebased. Thanks for the report!